workbench-core/base/src/cognitoTokenService.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import crypto from 'crypto';
+import AwsService from './aws/awsService';
+
+export default class CognitoTokenService {
+  private _aws: AwsService;
+
+  public constructor(awsRegion: string) {
+    this._aws = new AwsService({ region: awsRegion });
+  }
+
+  public async generateCognitoToken(
+    userPoolId: string,
+    clientId: string,
+    rootUsername: string,
+    rootPasswordParamStorePath?: string,
+    rootPassword?: string
+  ): Promise<{
+    accessToken: string;
+    idToken: string;
+    refreshToken: string;
+  }> {
+    let password: string = rootPassword || '';
+    if (rootPasswordParamStorePath && rootPassword) {
```

Comment:
throw error if `rootPassword` is defined, irrespective of `rootPasswordParamStorePath`